### DISCUSSION TITLE
모달 on/off를 위해 z-index를 활용하는 것에 대해 어떻게 생각하시나요?

### DISCUSSION THREAD

**Discussion by byhhh2** [woowacourse-study/2022-code-review-study#21] [Lotto] [answered]
[PR 120](https://github.com/woowacourse/javascript-lotto/pull/120#discussion_r817585439)을 보면 reflow나 repaint를 최대한 방지하기 위하여 z-index값을 이용하여 모달을 숨기고 보여주셨다고 적혀있습니다. 이 방법에 대해서 어떻게 생각하시나요?

**Comment by lokba** [comment, marked as answer]
오.. 일단 좋은 주제네요. 벌써부터 최적화를 고려하는 크루분이 계신군요.

저는 여러가지 측면으로 바라 보았어요.


on/off 대상이 스크린리더에 미리 포함되면 안되는 상황
-> z-index방식보다 display:none이 올바르다고 생각이 듭니다.

로또 미션처럼, 모달창이 on/off 대상인 경우 && 프로젝트 규모가 small
-> 프로젝트 규모가 작다면, display:none을 사용하겠습니다. 성능이 그렇게 중요하지 않기때문입니다.


로또 미션처럼, 모달창이 on/off 대상인 경우 && 프로젝트 규모가 big
-> 프로젝트 규모가 크다면, z-index를 사용할 거 같습니다. 제가 알고있는 지식으로는, 
display:none은 해당 element를 지우고 그려야 하기에, reflow 과정과 repaint 과정 둘다 진행하게 됩니다.
반면 z-index는 reflow 즉, 레이아웃이 변경되는 작업이 아니기에 repaint 과정만 진행하게 됩니다.

프로젝트 규모가 큰상황에서는 리소스의 절약이 중요하기에, 소소한 곳에서라도 리소스를 최소화하는게 좋을 것 같아 보이네요ㅎㅎ

**Reply by wonsss** [reply]
z-index가 display:none보다 성능상 더 유리하군요
몰랐던 내용인데, 무비와 록바 덕분에 알게 됐습니다. 감사해요~

다만, 한 가지 궁금한 점은 display:none은 웹접근성을 고려한 숨김처리라고 알고 있어요.
none을 주면 영역에서 아예 사라지니, 스크린리더 사용자는 해당 요소를 불필요하게 인지하지 않게 되는거죠.

그런데 z-index를 음수로 주는 방법은 영역에서 아예 사라지는 것이 아니니까, 화면에 보여지진 않더라도 스크린리더 사용자에게는
불필요하게 읽히게 되는 문제가 있지 않을까요?

추가로 웹 접근성을 고려하여 요소 숨기는 좋은 글이 있어 남기고 갑니다.
[웹 접근성을 고려하여 텍스트 숨기기](https://velog.io/@ursr0706/%EC%9B%B9-%EC%A0%91%EA%B7%BC%EC%84%B1%EC%9D%84-%EA%B3%A0%EB%A0%A4%ED%95%98%EC%97%AC-%ED%85%8D%EC%8A%A4%ED%8A%B8-%EC%88%A8%EA%B8%B0%EA%B8%B0)

**Reply by wonsss** [reply]
https://www.inflearn.com/course/%EC%9B%B9-%EC%84%B1%EB%8A%A5-%EC%B5%9C%EC%A0%81%ED%99%94-%EB%A6%AC%EC%95%A1%ED%8A%B8-1#curriculum
